File: A2/HobbyAnimals/Animals/Cat.cs
namespace HobbyAnimals;

//Primary Constructor inherited from Animal class
public class Cat(string name, int exhilaration, char type) : Animal(name, exhilaration, type)
{
    public override void changeEx(Mood mood)
    {
        switch (mood)
        {
            case Mood.joyful:
                increaseEx(3);
                break;
            case Mood.usual:
                increaseEx(3);
                break;
            case Mood.blue:
                decreaseEx(7);
                break;
        }
    }
}

File: A2_py/Animals/Cat.py
class Cat(Animal):
    def __init__(self, name: str, ex: int, type:str):
        self.name = name
        self.ex = ex
        self.type = type
    def changeEx(self, mood : Mood):
        match mood:
            case Mood.JOYFUL:
                self._increaseEx(3)
            case Mood.USUAL:
                self._increaseEx(3)
            case Mood.BLUE:
                self._decreaseEX(7)
